string key
string directory
---
string id
int32 num